pkg/kms/kms.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Chainguard, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kms
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	kmsGCP "cloud.google.com/go/kms/apiv1"
+	"github.com/aws/aws-sdk-go-v2/config"
+	kmsAWS "github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"github.com/octo-sts/app/pkg/kms/aws"
+	"github.com/octo-sts/app/pkg/kms/gcp"
+)
+
+const (
+	AWS = "aws"
+	GCP = "gcp"
+)
+
+type kmsProvider struct {
+	ctx       context.Context
+	provider  string
+	kmsKey    string
+	gcpClient *kmsGCP.KeyManagementClient
+	awsClient *kmsAWS.Client
+}
```

Comment:
Instead of storing all of the clients here, have each impl have a function that returns a struct that implements the KMS interface. Then we can do something like:

```
func NewKMS(ctx context.Context, provider, kmsKey string) (KMS, error) {
	switch strings.ToLower(provider) {
	case GCP:
		return gcp.NewProvider(ctx, kmsKey)
	case AWS:
		return aws.NewProvider(ctx, kmsKey)
	default:
		return nil, errors.New("unsupported kms provider")
	}
}
```

Alternatively, skip the KMS interface altogether and create funcs along the lines of `gcp.NewDefault(ctx, key)` that do the same thing here with default clients initialized.